src/main/java/org/gaul/httpbin/HttpBin.java

```diff
@@ -17,36 +17,72 @@
 
 package org.gaul.httpbin;
 
-import static java.util.Objects.requireNonNull;
-
-import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
 
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
 
 /**
  * Reimplementation of HttpBin https://httpbin.org/ suitable for offline unit
  * tests.
  */
 public final class HttpBin {
     private final Server server;
+    private final int mHTTPPort;
+    private final int mHTTPsPort;
 
-    public HttpBin(URI endpoint) throws Exception {
-        this(endpoint, new HttpBinHandler());
+    public HttpBin(String ip, int httpPort, int httpsPort, String keystore) throws Exception {
+        this(ip, httpPort, httpsPort, keystore, new HttpBinHandler());
     }
 
-    public HttpBin(URI endpoint, HttpBinHandler handler) throws Exception {
-        requireNonNull(endpoint);
+    public HttpBin(String ip, int httpPort, int httpsPort, String keystore, HttpBinHandler handler) throws Exception {
 
         server = new Server();
         HttpConnectionFactory httpConnectionFactory =
                 new HttpConnectionFactory();
-        ServerConnector connector = new ServerConnector(server,
+
+        mHTTPPort = httpPort;
+        mHTTPsPort = httpsPort;
+        List<Connector> connectors = new ArrayList<Connector>();
+        if (httpPort != 0) {
```

Comment:
Does it make sense to support the 0 port for automatic assignment and use something like -1 to mean do not use HTTP?